test/test_autograd.py

```diff
@@ -1093,6 +1093,15 @@ def test_requires_grad_factory(self):
                             self.assertIs(dtype, output.dtype)
                             self.assertEqual(1, output.get_device())
 
+    def test_attribute_deletion(self):
+        x = torch.randn((5, 5), requires_grad=True)
+        del x.grad
+        self.assertIsNone(x.grad)
+        del x.data
+        self.assertIsNone(x.data)
```

Comment:
What does it mean for a Tensor `.data` field to be `None` ? What happens to `x` when you do `del x.data` ? Does it become unusable?

Reply:
From my point of view, not allowing it would be better as this is a state you never want to be in (in python).